tests/test_server/test_endpoints.py

```diff
@@ -0,0 +1,200 @@
+            add_invocations_endpoint=True,
+        )
+        assert app.routes[-1].path == "/invocations"
+
+    def test_add_endpoint_with_no_route_specified(self, app):
+        _add_pipeline_endpoint(
+            app,
+            endpoint_config=Mock(endpoint=None),
+            pipeline=Mock(input_schema=StrSchema, output_schema=int),
+        )
+        assert app.routes[-1].path == "/predict"
+
+
+class TestActualModelEndpoints:
+    @pytest.fixture(scope="class")
+    def client(self):
+        server_config = ServerConfig(
+            num_cores=1,
+            num_workers=1,
+            endpoints=[
+                EndpointConfig(
+                    name="test endpoint 1",
+                    endpoint="/predict/static-batch",
+                    task="text-classification",
+                    model=Pipeline.default_model_for("text-classification"),
+                    batch_size=2,
+                ),
+                # TODO add these back in
+                # EndpointConfig(
+                #     name="test endpoint 2",
+                #     endpoint="/predict/dynamic-batch",
+                #     task="text-classification",
+                #     model=Pipeline.default_model_for("text-classification"),
+                #     batch_size=1,
+                # ),
+                # EndpointConfig(
+                #     name="test endpoint 3",
+                #     endpoint="/predict/bucketed",
+                #     task="text-classification",
+                #     model=Pipeline.default_model_for("text-classification"),
+                #     batch_size=1,
+                #     bucketing=SequenceLengthsConfig(sequence_lengths=[2, 4]),
+                # ),
+            ],
+        )
+        app = _build_app(server_config)
+        yield TestClient(app)
+
+    def test_static_batch_request(self, client):
+        response = client.post(
+            "/predict/static-batch",
+            json={
+                "sequences": [
+                    "today is great",
+                    "today is terrible",
+                ]
+            },
+        )
+        assert response.status_code == 200
+        assert response.json() == {
+            "labels": ["LABEL_1", "LABEL_0"],
+            "scores": [0.9998027682304382, 0.9995161890983582],
+        }
+
+    # def test_dynamic_batch_request(self, client):
+    #     response = client.post(
+    #         "/predict/static-batch",
+    #         json={
+    #             "sequences": [
+    #                 "today is great",
+    #                 "today is terrible",
+    #                 "today is great",
+    #             ]
+    #         },
+    #     )
+    #     assert response.status_code == 200
+    #     assert response.json() == {
+    #         "labels": ["LABEL_1", "LABEL_0", "LABEL_1"],
+    #         "scores": [0.9998027682304382, 0.9995161890983582, 0.9998027682304382],
+    #     }
+
+    # def test_bucketed_request(self, client):
+    #     ...
```

Comment:
remove or add back in?

Reply:
Will add back in thanks

Reply:
Okay so I'm taking out bucketing & dynamic batch testing at this server level. dynamic batch test is below, but will require the #604 PR to be merged. Bucketing seems to have some issues so we can address that and add tests in a separate PR.

```python
                EndpointConfig(
                    name="test endpoint 3",
                    endpoint="/predict/bucketed",
                    task="text-classification",
                    model=stub,
                    batch_size=1,
                    bucketing=SequenceLengthsConfig(sequence_lengths=[2, 4]),
                ),

    def test_dynamic_batch_handles_any_batch_size(self, client):
        sentences = [
            "today is great",
            "today is terrible",
            "today is average",
            "just okay today, thanks",
            "is this thing working??",
        ]
        for num_items in range(1, len(sentences) + 1):
            response = client.post(
                "/predict/dynamic-batch",
                json={"sequences": sentences[:num_items]},
            )
            assert response.status_code == 200
            output = response.json()
            assert len(output["labels"]) == num_items
            assert len(output["scores"]) == num_items
```

Reply:
Added a task in asana for this